shared/actions/chat2/index.tsx

```diff
@@ -1115,32 +1131,35 @@ function* loadMoreMessages(
 
   const loadingKey = Constants.waitingKeyThreadLoad(conversationIDKey)
   let calledClear = false
-  const onGotThread = (thread: string) => {
-    if (!thread) {
-      return
-    }
+  const onGotThread = (thread: string) =>
```

Comment:
this fixes a kinda subtle bad behavior which is async callbacks plumbed through `incomingCallMap` that access a stale `state` from their parent. The fact that this was plumbed through `state` being passed to these message factories i think obfuscated it further